src/test/ipv6_format.cpp

```diff
@@ -0,0 +1,36 @@
+#include "gtest/gtest.h"
+
+#include "../ipv6.hpp"
+#include <stdplus/tab.hpp>
+
+/* Frame (104 bytes) */
+static unsigned char icmp_frame00[104] = {
+0x60, 0x00, 0x00, 0x00, 0x00, 0x40, 0x3a, 0x2a, /* `....@:* */
+0xfc, 0xff, 0xc8, 0xa9, 0x4c, 0xd6, 0x58, 0x2b, /* ....L.X+ */
+0x32, 0x94, 0x05, 0x7f, 0x1d, 0xc5, 0x88, 0x8e, /* 2....... */
+0xfc, 0x1c, 0xb9, 0xe7, 0x34, 0x2d, 0xbb, 0x4c, /* ....4-.L */
+0xa6, 0x79, 0x04, 0xb2, 0x47, 0xac, 0x67, 0xb4, /* .y..G.g. */
+0x81, 0x00, 0x5e, 0x2e, 0x66, 0xc1, 0x00, 0x08, /* ..^.f... */
+0xcb, 0xf2, 0x2b, 0x59, 0x00, 0x00, 0x00, 0x00, /* ..+Y.... */
+0xb9, 0x51, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, /* .Q...... */
+0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, /* ........ */
+0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f, /* ........ */
+0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, /*  !"#$%&' */
+0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f, /* ()*+,-./ */
+0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37  /* 01234567 */
+};
+
+
+TEST(ipv6_format, payload_size) {
+	auto payload_size = 64;
+	stdplus::tab_view<unsigned char> icmp_view(icmp_frame00, icmp_frame00 + sizeof(icmp_frame00) );
```

Comment:
sizeof(icmp_frame00)

better use std::extent to get number of elements in array